src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -1627,6 +1627,16 @@ internal override System.Reflection.MethodImplAttributes ImplementationAttribute
 
         #endregion
 
+        internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions, DiagnosticBag diagnostics)
+        {
+            base.AfterAddingTypeMembersChecks(conversions, diagnostics);
+
+            if (IsDeclaredReadOnly)
+            {
+                this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(diagnostics, locations[0], modifyCompilation: true);
```

Comment:
>EnsureIsReadOnlyAttributeExists [](start = 42, length = 31)

looks like we were previously not emitting the readonly attribute in some scenarios. Can you add test to cover the change?

Reply:
This change is more or less consolidating the call to `EnsureIsReadOnlyAttributeExists` into `SourceMemberMethodSymbol` instead of doing it multiple times in `SourceOrdinaryMethodSymbol` and `SourcePropertyAccessorSymbol`.

See https://github.com/dotnet/roslyn/pull/34778/files#diff-e41978e6aacc94dab61ff4ccda181885L1141 and https://github.com/dotnet/roslyn/pull/34778/files#diff-2d35c3c328d688de341a0a23f05b37bdL624

Reply:
It's been a little bit but I think this may have closed over a gap where emit would fail for `readonly event` because we forgot to `EnsureIsReadOnlyAttributeExists` in the old factoring. So I'll make sure that test is covered.

Reply:
Done